### PR TITLE
LWC and Aura Component to create new Lead functionality

### PR DESCRIPTION
**Pull request by kaurarjinder**
# Critical Changes

# Changes

# Issues Closed

### CODE REVIEW

File: force-app/main/default/lwc/newLeadCmp/newLeadCmp.html [hunk 1]
```diff
@@ -0,0 +1,57 @@
+<template>
+    <c-modal-cmp title="New Lead" onclose={handleCancel} onsubmit={handleSubmit}>
```
Comment (deb761) [inline force-app/main/default/lwc/newLeadCmp/newLeadCmp.html:2]: The form looks great.  Thanks for putting it together.  The onerror doesn't seem to be needed.  I'd see what @atlasjen thinks on that.  We'd want the new contact form and new lead form to act the same.

image

File: force-app/main/default/lwc/newLeadCmp/newLeadCmp.js-meta.xml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<LightningComponentBundle xmlns="http://soap.sforce.com/2006/04/metadata">
+    <apiVersion>51.0</apiVersion>
+    <isExposed>true</isExposed>
+    <targets>
+        <target>lightning__HomePage</target>
```
[the remark this answers is not in the document]
Comment (atlasjen) [reply force-app/main/default/lwc/newLeadCmp/newLeadCmp.js-meta.xml:6]: New Lead looks really nice.
- minor changes. Phone is not required.
- If click "specify" on pronoun, it should open a new box
- Maybe move the interest to its own line so can see the full text
- weren't we going to make Company "N/A" by default and autopopulate?
- change "Description Information" to be "Comment or Message" and title the box itself to be "Comment"


Is it possible to have validation flag all fields vs just ones previously clicked on or 1st one? That seems to be the behavior?

Popup is good, can we change the text to remove "below" in the error?

**Comment by atlasjen** [comment]
@kaurarjinder I made a couple of small tweaks and checked them in, including adding the Lead form to the 'new' button. I think we are really close :) I like how you have organized what goes where on each form.
**For New Contact:**

- Need a field to display if they choose "specify" for pronoun so they can write something in
- missing Lead Interest (multiselect)
- for Lead source
--need another field to display if they choose "Other"  so they have a field to write in
--should be multi select
- it takes a long time for the model to close after the popup that shows the contact was created. can we change that delay at all?


**For New Lead:**

- missing pronoun field, and will also need to deal with if they choose "specify" for pronoun so they have a field to write in
- missing Lead Interest (multiselect)
- for Lead source
-- need another field to display if they choose "Other"  so they have a field to write in
--should be multi select
- it takes a long time for the model to close after the popup that shows the contact was created. can we change that delay at all?

**Comment by kaurarjinder** [comment]
Hi Jen,

Specify / Other fields and lead interest fields which you have mentioned
should be visible to you.
I have added those fields. Those fields were visible to me when I did the
testing.
I will check again.
I will see how to make it fast and avoid delay.

Thanks

**Pull request merged**